packages/compass-query-bar/src/components/query-bar/query-bar.jsx

```diff
@@ -220,6 +223,7 @@ class QueryBar extends Component {
       this.props[option] : this.props[`${option}String`];
 
     const label = OPTION_DEFINITION[option].label || option;
+    const placeholder = option === 'sort' && !supportSquareBracketSorts ? OPTION_DEFINITION[option].nonSquareBracketPlaceholder : OPTION_DEFINITION[option].placeholder;
```

Comment:
I don't think it's the approach to allow for the behavior you want, this api feels very specific and hard to scale (what if next you would want to change a label on another field, how many of those custom props for any of them will end up in this component interface?). Maybe something more generic that treats `OPTION_DEFINITION` as default options and allows to override them with props might be more appropriate here

Reply:
Got it! Will fix this and push again